pkg/daemon/daemon.go

```diff
@@ -124,6 +124,11 @@ type Daemon struct {
 	drainer *drain.Helper
 }
 
+// CoreOSDaemon protects the methods that should only be called on CoreOS variants
+type CoreOSDaemon struct {
+	*Daemon
```

Comment:
I think longer term what we want is to drop the `*` here - but it looks like that would be a large refactoring which can come later.

(i.e. we early on in MCD startup detect if it's coreos and construct a `CoreOSDaemon` instead of a raw `Daemon` in `runStartCmd()` )